tests/components/subaru/test_config_flow.py

```diff
@@ -0,0 +1,250 @@
+    ) as mock_update_saved_pin:
+        result = await hass.config_entries.flow.async_configure(
+            pin_form["flow_id"], user_input={CONF_PIN: TEST_PIN}
+        )
+
+    assert len(mock_test_pin.mock_calls) == 1
+    assert len(mock_update_saved_pin.mock_calls) == 1
+    expected = {
+        "title": TEST_USERNAME,
+        "description": None,
+        "description_placeholders": None,
+        "flow_id": mock.ANY,
+        "result": mock.ANY,
+        "handler": DOMAIN,
+        "type": "create_entry",
+        "version": 1,
+        "data": TEST_CONFIG,
+    }
+    result["data"][CONF_DEVICE_ID] = TEST_DEVICE_ID
+    assert result == expected
+
+
+async def test_pin_form_incorrect_pin(hass, pin_form):
+    """Test we handle invalid pin."""
+    with patch(
+        MOCK_API_TEST_PIN,
+        side_effect=InvalidPIN("invalidPin"),
+    ) as mock_test_pin, patch(
+        MOCK_API_UPDATE_SAVED_PIN,
+        return_value=True,
+    ) as mock_update_saved_pin:
+        result = await hass.config_entries.flow.async_configure(
+            pin_form["flow_id"], user_input={CONF_PIN: TEST_PIN}
+        )
+    assert len(mock_test_pin.mock_calls) == 1
+    assert len(mock_update_saved_pin.mock_calls) == 1
+    assert result["type"] == "form"
+    assert result["errors"] == {"base": "incorrect_pin"}
+
+
+async def test_option_flow_form(options_form):
+    """Test config flow options form."""
+    expected = {
+        "data_schema": mock.ANY,
+        "description_placeholders": None,
+        "errors": None,
+        "flow_id": mock.ANY,
+        "handler": mock.ANY,
+        "step_id": "init",
+        "type": "form",
+    }
+    assert expected == options_form
+
+
+async def test_option_flow(hass, options_form):
+    """Test config flow options."""
+    result = await hass.config_entries.options.async_configure(
+        options_form["flow_id"],
+        user_input={
+            CONF_UPDATE_ENABLED: False,
+        },
+    )
+    assert result["type"] == "create_entry"
+    assert result["data"] == {
+        CONF_UPDATE_ENABLED: False,
+    }
+
+
+@pytest.fixture
+async def user_form(hass):
+    """Return initial form for Subaru config flow."""
+    return await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+
+@pytest.fixture
+async def pin_form(hass, user_form):
+    """Return second form (PIN input) for Subaru config flow."""
+    with patch(MOCK_API_CONNECT, return_value=True,), patch(
+        MOCK_API_IS_PIN_REQUIRED,
+        return_value=True,
+    ):
+        return await hass.config_entries.flow.async_configure(
+            user_form["flow_id"], user_input=TEST_CREDS
+        )
+
+
+@pytest.fixture
+async def options_form(hass):
+    """Return options form for Subaru config flow."""
+    entry = MockConfigEntry(domain=DOMAIN, data={}, options=None)
+    entry.add_to_hass(hass)
+    return await hass.config_entries.options.async_init(entry.entry_id)
```

Comment:
We must set up the integration before starting the options flow in tests to make sure that the config flow module is loaded.